Add 1.18 and Below Support, and more

MrFawkes1337 opened the pull request:
This PR adds support for 1.17+ Mobs (Goats, Axolotl, Glow Squid), Fixes the console spam errors relating to Cats being Tamed, Separates the mobs into Hostile, Passive, and More Creatures Layers, which requires a clean config.yml to work. Gradle support updated to 7.4 for Compiling, while also updating the Maven plugins, Spigot, Dynmap API to 3.2, as Markers will need a rewrite to support the new format. a Compiled lib can be found at https://www.github.com/MrFawkes1337/dynmap-mobs/releases/ and seems to work well on testing on my server, which I can also provide if requested. This PR aims to include ALL changes others have made, as well as my own for updated support.

Commit:
Update to 1.13.2

Commit:
Show custom name

gabsoftware committed:
Minecraft 1.14.x mobs compatibility

gabsoftware committed:
Minecraft 1.14.x mobs compatibility (#1)

gabsoftware committed:
improved support for 1.14.x (villager professions, rideable mobs...)

gabsoftware committed:
comments

gabsoftware committed:
improved compatibility (#2)

* Minecraft 1.14.x mobs compatibility

* improved support for 1.14.x (villager professions, rideable mobs...)

* comments

gabsoftware committed:
splitted mobs layer into Mo'Creatures, hostile and passive mobs layers

gabsoftware committed:
Merge branch 'master' into master

gabsoftware committed:
1.15.1 compatibility with Bee support

gabsoftware committed:
1.15.1 compatibility with Bee support #2

pakfield committed:
Added 1.16 mobs

Commit:
Add 1.17 Mobs

Adds Mobs added with 1.17 to Dynmap Mobs

Commit:
Update versions

Update to Latest Spigot Api, Also up the Version to reflect new changes

Commit:
Update Config.yml with 1.17 Mobs

Add 1.17 Mobs support to Dynmap-Mobs

Commit:
Add Images for Mobs

Adds Images for Mobs added in 1.17

Commit:
Update Icons with Correct Ones

Add correct Icons

Commit:
Add 8x8 Icons

Adds 8x8 Versions of the Icons

Commit:
Update to include support for 1.17+ Mobs

Adds support for 1.17+ Mobs, Compiled Plugin located in Releases, labelled Dynmap-mobs-1.7.jar

Commit:
Add full support for 1.18+

This adds support for all versions below 1.18, with New Mobs for all. Not sure if bugs, will be stress tested

Commit:
Rename Vanilla Goat to fix compat issue

Fixes compatibility between Mo Mobs Goats and Vanilla 1.17 Goats. Renames the 1.17 Goat to vanillagoat

Commit:
Update pom.xml

Commit:
Clean Repo of Gradle

Remove gradle and all files completely, Cleans everything back to just a Compile with Maven as it should be.

ghost commented:
Latest PR Completely removes Gradle, reverting project back to a Maven package as intended, I tested with 3.4 API, Updated the location for Spigot API and all seems to work fine, including versioning which with gradle didn't for some reason. Includes Layers for Hostile, Passive, Mo Mobs, 1.17 and Below mob support.

Commit:
Reapply Vanilla Goat Fix

Reapply Vanilla Goat Fix, as I accidentally removed the Icons, and reset Config.

Commit:
Update MetricsLite

Updates MetricsLite to the latest Version.

Commit:
Add api-version to plugin

Adds the 1.18 Api-version to the Plugin.yml to fix Bukkits Legacy content warning.

Commit:
Update DynmapMobsPlugin.java

Commit:
Update pom.xml

Update to using 3.4 API Snapshot. Maven Compiles fine, meaning it was a gradle issue.

Commit:
Remove mcstats, add bstats

Removes the old Mcstats MetricsLite, and adds BStats as a Dependency to compile, removing Metrics completrely

Commit:
Delete target/classes directory

Commit:
Create Maven Build file

Add Auto Build for New Commits, this is a test. Probably won't work

Commit:
Create maven.yml

Commit:
Delete maven-publish.yml

Commit:
Update DynmapMobsPlugin.java

Remove mistake left in place of MetricsLite

Commit:
Update maven.yml

Plastikmensch committed:
fix: show tamed mobs on map

Plastikmensch committed:
fix: add and check for tamed mobs

Plastikmensch committed:
fix: add new config keys

Plastikmensch committed:
fix: add new icons

Plastikmensch committed:
chore: version bump

Plastikmensch committed:
fix: distuingish between llamas and trader llamas

Commit:
Merge pull request #1 from Plastikmensch/fix-tamed

fix: show tamed mobs on map, Llama distinguish, and more

Commit:
Update maven.yml
